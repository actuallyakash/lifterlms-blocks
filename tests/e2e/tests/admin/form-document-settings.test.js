/**
 * Test Form post document sidebar
 *
 * @since [version]
 * @version [version]
 */

import {
	clickElementByText,
	toggleOpenRegistration,
} from '@lifterlms/llms-e2e-test-utils';

import {
	visitForm,
} from '../../util';

async function openFormSettingsPanel() {

	const isOpen = await page.$eval( '.llms-forms-doc-settings', el => el.classList.contains( 'is-opened' ) );

	if ( ! isOpen ) {
		await clickElementByText( 'Form Settings', '.components-panel .components-button' );
	}

}

describe( 'Admin/FormsDocSidebar', () => {

	it ( 'should display the location when open registration is disabled', async () => {

		await toggleOpenRegistration( false );

		await visitForm( 'Register' );

		await openFormSettingsPanel();

		expect( await page.$eval( '.llms-forms-doc-settings .components-panel__row', el => el.innerHTML ) ).toMatchSnapshot();

	} );

	it ( 'should display the location with a link when open registration is enabled', async () => {

		await toggleOpenRegistration( true );

		await visitForm( 'Register' );

		await openFormSettingsPanel();

		expect( await page.$eval( '.llms-forms-doc-settings .components-panel__row', el => el.innerHTML ) ).toMatchSnapshot();

	} );

} );
