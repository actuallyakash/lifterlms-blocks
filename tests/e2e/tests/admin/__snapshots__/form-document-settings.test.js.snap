// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin/FormsDocSidebar should display the location when open registration is disabled 1`] = `"<strong>Location</strong><strong>Registration</strong>"`;

exports[`Admin/FormsDocSidebar should display the location with a link when open registration is enabled 1`] = `"<strong>Location</strong><a class=\\"components-external-link\\" href=\\"http://localhost:8484/dashboard/\\" target=\\"_blank\\" rel=\\"external noreferrer noopener\\">Registration<span class=\\"components-visually-hidden\\">(opens in a new tab)</span><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" width=\\"24\\" height=\\"24\\" class=\\"components-external-link__icon css-6wogo1-StyledIcon etxm6pv0\\" role=\\"img\\" aria-hidden=\\"true\\" focusable=\\"false\\"><path d=\\"M18.2 17c0 .7-.6 1.2-1.2 1.2H7c-.7 0-1.2-.6-1.2-1.2V7c0-.7.6-1.2 1.2-1.2h3.2V4.2H7C5.5 4.2 4.2 5.5 4.2 7v10c0 1.5 1.2 2.8 2.8 2.8h10c1.5 0 2.8-1.2 2.8-2.8v-3.6h-1.5V17zM14.9 3v1.5h3.7l-6.4 6.4 1.1 1.1 6.4-6.4v3.7h1.5V3h-6.3z\\"></path></svg></a>"`;
